plugins/inputs/dcos_containers/dcos_containers.go

```diff
@@ -136,6 +136,22 @@ func (dc *DCOSContainers) getContainers(ctx context.Context, cli calls.Sender) (
 	return gc, nil
 }
 
+type roundTripper struct {
+	r         http.RoundTripper
+	userAgent string
+}
+
+var defaultUserAgent = "telegraf-dcos-containers"
+
+// RoundTrip is an implementation of the RoundTripper interface.
+func (rt roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", defaultUserAgent)
+	if rt.userAgent != "" {
+		req.Header.Set("User-Agent", rt.userAgent)
+	}
+	return rt.r.RoundTrip(req)
+}
```

Comment:
The `roundTripper` struct and its `RoundTrip` method are duplicated in each plugin. How about we extract it to a library at the top level of this project, alongside `testutil` etc.?

Reply:
Had discussed the possibility of extracting common logic out previously (there is also duplicate logic with IAM handling; had created [`DCOSConfig` thinking this would be eventually extracted out](https://github.com/dcos/telegraf/pull/23#issue-220221356)), I'll take a stab at trying to do that in this PR.

Reply:
I've created `dcosutil`